src/Util.cc

```diff
@@ -216,5 +216,31 @@ void pinAvailableCore() {
     pinThreadToCore(coreId);
 }
 
+/**
+  * Returns the CPU ID of coreId's hypertwin.  If there is no hypertwin, return -1.
+  * The returned ID matches what would be returned by a thread on the hypertwin
+  * (if a hypertwin exists) that ran sched_getcpu().
+  *
+  * \param coreId
+  *     The coreId whose hypertwin's ID will be returned.
+  */
+int getHyperTwin(int coreId) {
```

Comment:
Isn't this going to have bad performance consequences? This gets invoked in the inner loop of the CoreArbiter's scheduler, no? I would have expected that you would read all the data once, keep a copy in memory, and serve future requests instantaneously using that copy. Let's discuss.

Reply:
I've already done this--just in the core arbiter instead of in perfutils.  I kept the perfutils function deliberately general because I suspect it will be needed elsewhere.